libraries/integration/Microsoft.Bot.Builder.Integration.AspNet.Core/ServiceCollectionExtensions.cs

```diff
@@ -0,0 +1,110 @@
+        public static void AddBotRuntime(this IServiceCollection services, IConfiguration configuration)
+        {
+            RegisterCommonServices(services, configuration);
+        }
+
+        /// <summary>
+        /// Overload that allows for creating the Bot (IBot) type as well as other startup types.
+        /// </summary>
+        /// <typeparam name="TBot">Type of Bot.</typeparam>
+        /// <param name="services">The application's collection of registered services.</param>
+        /// <param name="configuration">The application configuration.</param>
+        public static void AddBotRuntime<TBot>(this IServiceCollection services, IConfiguration configuration) 
+            where TBot : class, IBot
+        {
+            RegisterCommonServices(services, configuration);
+
+            services.AddTransient<IBot, TBot>();
+        }
+
+        /// <summary>
+        /// Overload that allows for providing the Bot (IBot) type, and the main Dialog type as well as other startup types.
+        /// </summary>
+        /// <typeparam name="TBot">Type of Bot.</typeparam>
+        /// <typeparam name="TDialog">Type of Dialog.</typeparam>
+        /// <param name="services">The application's collection of registered services.</param>
+        /// <param name="configuration">The application configuration.</param>
+        public static void AddBotRuntime<TBot, TDialog>(this IServiceCollection services, IConfiguration configuration)
+            where TBot : class, IBot
+            where TDialog : Dialog
+        {
+            RegisterCommonServices(services, configuration);
+            services.AddSingleton<TDialog>();
+            services.AddTransient<IBot, TBot>();
+        }
+
+        private static void RegisterCommonServices(IServiceCollection services, IConfiguration configuration)
+        {
+            if (services == null)
+            {
+                throw new ArgumentNullException(nameof(services));
+            }
+
+            if (configuration == null)
+            {
+                throw new ArgumentNullException(nameof(configuration));
+            }
+
+            // Ensure the IConfiguration is available. (Azure Functions don't do this.)
+            services.TryAddSingleton(configuration);
+
+            // All things auth
+            services.TryAddSingleton<BotFrameworkAuthentication, ConfigurationBotFrameworkAuthentication>();
+
+            // Check to see if a different IBotFrameworkHttpAdapter was already registered, if it was, skip registering.
+            // We do this not because of the IBotFrameworkHttpAdapter, but to keep from adding the CoreBotAdapter if 
+            // the bot won't be using it and a different implementation was registered.
+            if (!services.Any(x => x.ServiceType == typeof(IBotFrameworkHttpAdapter)))
+            {
+                // CoreBotAdapter registration
+                services.TryAddSingleton<CoreBotAdapter>();
```

Comment:
Is this necessary? Is there ever a time where CoreBotAdapter needs to be retrieved from di directly as CoreBotAdapter?

CoreBotAdapter could be internal and hidden in this assembly, and registration could just be:
`services.TryAddSingleton<IBotFrameworkHttpAdapter, CoreBotAdapter >();`
adding the hidden CoreBotAdapter if there is not already a registered IBotFrameworkHttpAdapter

Reply:
Good suggestion, this may have been carried over from the other implementation and isn't needed here.